lib/imports.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+const {getOriginal} = require('./originals-helper.js');
+
+/**
+ * Formats the information into the right format for import
+ *
+ * @param {string} identifier - the name of the method / function to get
+ * @param {string} type - the type of original of the identifier
+ * @param {string} mod - the name of the module to import from
+ *
+ * @return {string} the formatted import text
+ */
+function generateImportText(identifier, type, mod) {
+  const original = getOriginal(identifier, type, mod);
+  switch (type) {
+    case 'constructor':
+      return original;
+    case 'method':
+      return `${identifier} as ${original}`;
+    default:
+      throw new Error('not yet implemented');
+  }
+}
+
+/**
+ * Adds an import for the module to the list
+ *
+ * @param {string} mod - the name of the module to import from
+ * @param {string} identifier - the name of the method / function to get
+ * @param {string} type - the type of original of the identifier
+ * @param {Object} imports - the current list of imports
+ */
+function updateImports(mod, identifier, type, imports) {
+  const importText = generateImportText(identifier, type, mod);
+
+  // TODO(jackbsteinberg): update this for other call-requiring cases
+  // add original for `call` from `Reflect` upon first method call
+  if (type === 'method' && !imports.Reflect) {
+    imports.Reflect = {NAMED_IMPORTS: ['call as Reflect_call']};
+  }
+
+  if (mod in imports) {
+    if (type === 'constructor') {
+      imports[mod].DEFAULT = importText;
+      return;
+    }
+
+    if (identifier in imports[mod].NAMED_IMPORTS) {
```

Comment:
This will never be true, because NAMED_IMPORTS is an array, so it will only contain numeric keys.

This indicates this code path is not being tested. Write a failing test that exercises what you're trying to do here, then fix the code. (Writing the test first is pretty important here.)

Reply:
I will go ahead and fix this, but it actually doesn't seem to matter, the tests won't fail because somewhere in the process of emitting the project to a file in memory the duplicate imports get cleaned up. I added a line to another test that would test this if the emitting didn't just fix it for me